pkg/network/state.go

```diff
@@ -474,6 +481,37 @@ func (ns *networkState) storeHTTPStats(allStats map[http.Key]*http.RequestStats)
 	}
 }
 
+// storeHTTPStats stores the latest HTTP stats for all clients
+func (ns *networkState) storeHTTP2Stats(allStats map[http.Key]*http.RequestStats) {
+	if len(ns.clients) == 1 {
+		for _, client := range ns.clients {
+			if len(client.http2StatsDelta) == 0 {
+				// optimization for the common case:
+				// if there is only one client and no previous state, no memory allocation is needed
+				client.http2StatsDelta = allStats
+				return
+			}
+		}
+	}
+
+	for key, stats := range allStats {
+		for _, client := range ns.clients {
+			prevStats, ok := client.http2StatsDelta[key]
+			if !ok && len(client.http2StatsDelta) >= ns.maxHTTPStats {
+				ns.telemetry.httpStatsDropped++
+				continue
+			}
+
+			if prevStats != nil {
+				prevStats.CombineWith(stats)
+				client.httpStatsDelta[key] = prevStats
```

Comment:
`http2StatsDelta`? This code worries me, it is a copy of the `storeHTTPStats` with the exception of this map. Any way to you can have the two functions share functionality?

Reply:
Should this be `http2StatsDelta`?

Reply:
good catch

Reply:
correct - fixed! thanks!